### ISSUE TITLE
security-csrf 2.8 depends on unsupported & vulnerable security-core 3.0

### ISSUE THREAD

**Issue by stephank** [symfony/symfony#27507]
**Symfony version(s) affected**: 2.8

**Description**  
Installing version 2.8 of `symfony/security-csrf` by itself installs an unsupported and vulnerable version of `symfony/security-core`.

The following line causes composer to prefer the unsupported 3.0 version of `symfony/security-core`:
https://github.com/symfony/symfony/blob/efe9beb1863f6a80fb975200c8dd6227d8f2befc/src/Symfony/Component/Security/Csrf/composer.json#L22

The last released version on the 3.0 branch is 3.0.9, which is listed as vulnerable: https://github.com/FriendsOfPHP/security-advisories/blob/e9093b7fc2649d99119cf2540ee6a66471c32480/symfony/security-core/CVE-2018-11407.yaml

**How to reproduce** 
- Start a new project (`composer init`)
- `composer require 'symfony/security-csrf:^2.8'`

Result:
```
$ composer require 'symfony/security-csrf:^2.8'
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 6 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-util (v1.8.0): Loading from cache
  - Installing symfony/polyfill-php56 (v1.8.0): Loading from cache
  - Installing symfony/security-core (v3.0.9): Loading from cache
  - Installing paragonie/random_compat (v2.0.12): Loading from cache
  - Installing symfony/polyfill-php70 (v1.8.0): Loading from cache
  - Installing symfony/security-csrf (v2.8.41): Loading from cache
symfony/security-core suggests installing symfony/event-dispatcher ()
symfony/security-core suggests installing symfony/expression-language (For using the expression voter)
symfony/security-core suggests installing symfony/http-foundation ()
symfony/security-core suggests installing symfony/ldap (For using LDAP integration)
symfony/security-core suggests installing symfony/validator (For using the user password constraint)
paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
symfony/security-csrf suggests installing symfony/http-foundation (For using the class SessionTokenStorage.)
Writing lock file
Generating autoload files
```

**Possible Solution**  
Loosen the dep to allow any 3.x version or disallow the dep on 3.0, (but that may be backwards breaking.)

**Comment by stof** [comment]
If you want to stay on the 2.8 LTS while using component packages rather than the fullstack, I suggest you to install the symfony/lts package.

Our handling of non-fullstack projects regarding LTS was not that good for 2.8, due to the fact that 99% (that number is a guess, not an exact number) of Symfony projects were fullstack ones at that time.

and composer does not know the concept of maintained vs unmaintained version, so we cannot tell it to prefer 2.8.x versions over 3.0.x versions due to that.
When working on the 3.4 LTS, we thought about the issue of inter-component dependencies (as we were working on the Flex world where fullstack is not the main usage anymore), and this is why we changed our policy for our requirements and we created the `symfony/lts` package to enforce having *all* components on the LTS version.

And if you want prevent installing vulnerable versions during a composer update, you might also want to use https://packagist.org/packages/roave/security-advisories
